C++/Dijkstras_Algorithm/dijkstras_algorithm_sssp.cpp

```diff
@@ -0,0 +1,99 @@
+// Dijkstra's Algorithm (SSSP)
+// Weighted Graph (Directed OR Undirected)
+// Positive Weighted Edges. Fails For Negative Weights.
+// O(ELogV)
+#include <bits/stdc++.h>
+using namespace std;
+
+void init() {
+	ios_base::sync_with_stdio(false);
+	cin.tie(NULL); cout.tie(NULL);
+#ifndef ONLINE_JUDGE
+	freopen("input.txt", "r", stdin);
+	freopen("output.txt", "w", stdout);
+#endif
+}
+
+template <typename T>
+class Graph {
+	int V;
+	unordered_map<T, list<pair<T, int>>> um; // (string->(string,weight))
+	
+public:
+	Graph(int V) {
+		this->V = V;
+	}
+
+	void addEdge(T x, T y, int wt, bool bidir = true) {
+		um[x].push_back(make_pair(y, wt));
+		if (bidir) {
+			um[y].push_back(make_pair(x, wt));
+		}
+	}
+
+	void printAdjList() {
+		for (auto edge : um) {
+			T node = edge.first;
+			cout << node << "--->";
+			for (auto s : edge.second) {
+				cout << "(" << s.first << "," << s.second << ")";
+			}
+			cout << endl;
+		}
+	}
+
+	void dijkstraSSSP(T src) {
+		unordered_map<T, int> dist;
+		// Initially all the nodes are at infinite(INT_MAX) distance.
+		for (auto edge : um) {
+			T node = edge.first;
+			dist[node] = INT_MAX;
+		}
+		// Used to get minimum in O(1) time and ramove in O(logN) time.
+		set<pair<int, T>> s; // (dist,vertex)
```

Comment:
Bad coding practice, you are using a set instead of Queue or arraydeque, this is going to increase the complexity of that algorithm to more than worse complexity as erase() function causes costly rehashing.